src/lm.jl

```diff
@@ -200,19 +173,44 @@ deviance(obj::LinearModel) = deviance(obj.rr)
 
 For linear models, the deviance of the null model is equal to the total sum of squares (TSS).
 """
-nulldeviance(obj::LinearModel) = nulldeviance(obj.rr)
-loglikelihood(obj::LinearModel) = loglikelihood(obj.rr)
-nullloglikelihood(obj::LinearModel) = nullloglikelihood(obj.rr)
+function nulldeviance(obj::LinearModel) 
+    y = obj.rr.y
+    wts = obj.rr.wts
+    
+    if hasintercept(obj)
+        if isempty(wts)
+            m = mean(y)
+        else 
+            m = mean(y, weights(wts))
+        end
+    else
+        m = zero(eltype(y))
```

Comment:
OK, let's try something like this then? It would also be good to adapt Project.toml to require StatsAPI 1.4, which I'll tag shortly (https://github.com/JuliaStats/StatsAPI.jl/pull/14), so that the docstring is consistent with what we do here.
```suggestion
        @warn("Starting from GLM.jl 1.8, null model is defined as having no predictor at all " *
              "when a model without an intercept is passed.")
        m = zero(eltype(y))
```